CONTRIBUTING.md

```diff
@@ -0,0 +1,20 @@
+# Contributing
+
+## Formatting
+
+Please format your code before before opening pull requests using clang-format 11 and the .clang-format file placed in the repository root.
+
+### Visual Studio and CLion
+Suport for clang-format is built-in since Visual Studio 2017 15.7 and CLion 2019.1.
+The .clang-format file in the repository will be automatically detected and formatting is done as you type, or triggered when pressing the format hotkey.
+
+### Bash
+First install clang-format-11. Instructions therefore can be found on the web.
+To format your changes since branching off develop, you can run this command in bash:
+```
+git clang-format-11 develop
+```
+To format all code in your working copy, you can run this command in bash:
+```
+find -iname *.cpp -o -iname *.hpp | xargs clang-format-11 -i
```

Comment:
This command is wrong, it must be `find -iname "*.cpp" -o -iname "*.hpp" | xargs clang-format-11 -i`

Reply:
`*` is in the bash evaluated but this should define only a pattern for a file.

Reply:
Funnily enough, the command works just fine in my bash.

Reply:
Ohh let's check if this issue is coming from my `zsh`

Reply:
Mhh it is the zshell which do not like `*` only.

Reply:
Yes, `zsh` needs either the quotes or a backslash. I prefer the quotes because it is portable and looks okay.

Reply:
Very well then, you have my permission to fix that then :)

Reply:
I can do it tomorrow.